lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -321,9 +325,24 @@ def parse_element(raw, node)
         raw
       end
 
-      def commit(action, request, *success_criteria)
-        xmr = ssl_post(url, request, 'Content-Type' => 'text/xml', 'Authorization' => encoded_credentials)
+      def headers(options)
+        headers = {
+          'Content-Type' => 'text/xml',
+          'Authorization' => encoded_credentials
+        }
+        if options[:cookie]
+          headers.merge!('Set-Cookie' => options[:cookie])
```

Comment:
Minor tweak for consistency: `headers.merge!('Set-Cookie' => options[:cookie]) if options[:cookie]`.